mmv1/templates/terraform/custom_flatten/alloydb_cluster_input_maintenance_window_start_time_flatten.go.erb

```diff
@@ -0,0 +1,95 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2024 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+	if v == nil {
+		return v
+	}
+	original := v.(map[string]interface{})
+    transformed := map[string]interface{}{
+        "hours": flattenAlloydbClusterMaintenanceWindowStartTimeHours(original["hours"], d, config),
+        "minutes": flattenAlloydbClusterMaintenanceWindowStartTimeMinutes(original["minutes"], d, config),
+        "seconds": flattenAlloydbClusterMaintenanceWindowStartTimeSeconds(original["seconds"], d, config),
+        "nanos": flattenAlloydbClusterMaintenanceWindowStartTimeNanos(original["nanos"], d, config),
+    }
+	return []interface{}{transformed}
+}
+
+func flattenAlloydbClusterMaintenanceWindowStartTimeHours(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+    // Handles the string fixed64 format
```

Comment:
Is the response from the API different for an int vs a float? Wouldn't parsing as a float always work?

Reply:
From our API we should return int, float should always work.
I think you are saying the handling for string or float is not required here, right?

Reply:
Yeah, if it's always an int then our typical generated flatten methods should work. You should be able to remove the custom flatten entirely